code/game/machinery/doors/firedoor.dm

```diff
@@ -140,4 +176,148 @@
 	icon = 'icons/obj/doors/Doorfire.dmi'
 	glass = 0
 
+/obj/item/weapon/firelock_board
+	name = "firelock circuitry"
+	desc = "A circuit board used in construction of firelocks."
+	icon = 'icons/obj/module.dmi'
+	icon_state = "mainboard"
+	w_class = 2
+	materials = list(MAT_METAL = 50, MAT_GLASS = 50)
+
+/obj/structure/firelock_frame
+	name = "firelock frame"
+	desc = "A partially completed firelock."
+	icon = 'icons/obj/doors/Doorfire.dmi'
+	icon_state = "frame1"
+	anchored = 0
+	density = 1
+	var/constructionStep = CONSTRUCTION_NOCIRCUIT
+	var/reinforced = 0
+
+/obj/structure/firelock_frame/examine(mob/user)
+	..()
+	switch(constructionStep)
+		if(CONSTRUCTION_PANEL_OPEN)
+			user << "There is a small metal plate covering the wires."
+		if(CONSTRUCTION_WIRES_EXPOSED)
+			user << "Wires are trailing from the maintenance panel."
+		if(CONSTRUCTION_GUTTED)
+			user << "The circuit board is visible."
+		if(CONSTRUCTION_NOCIRCUIT)
+			user << "There are no electronics in the frame."
+
+/obj/structure/firelock_frame/update_icon()
+	..()
+	icon_state = "frame[constructionStep]"
 
+/obj/structure/firelock_frame/attackby(obj/item/weapon/C, mob/user)
+	switch(constructionStep)
+		if(CONSTRUCTION_PANEL_OPEN)
+			if(istype(C, /obj/item/weapon/crowbar))
+				playsound(get_turf(src), 'sound/items/Crowbar.ogg', 50, 1)
+				user.visible_message("<span class='notice'>[user] starts prying something out from [src]...</span>", \
+									 "<span class='notice'>You begin prying out the wire cover...</span>")
+				if(!do_after(user, 50)) return
+				playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 50, 1)
+				user.visible_message("<span class='notice'>[user] pries out a metal plate from [src], exposing the wires.</span>", \
+									 "<span class='notice'>You remove the cover plate from [src], exposing the wires.</span>")
+				constructionStep = CONSTRUCTION_WIRES_EXPOSED
+				update_icon()
+				return
+			if(istype(C, /obj/item/weapon/wrench))
+				playsound(get_turf(src), 'sound/items/Ratchet.ogg', 50, 1)
+				user.visible_message("<span class='notice'>[user] starts bolting down [src]...</span>", \
+									 "<span class='notice'>You begin bolting [src]...</span>")
+				if(!do_after(user, 30)) return
+				user.visible_message("<span class='notice'>[user] finishes the firelock.</span>", \
+									 "<span class='notice'>You finish the firelock.</span>")
+				playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 50, 1)
+				if(reinforced)
+					new /obj/machinery/door/firedoor/heavy(get_turf(src))
+				else
+					new /obj/machinery/door/firedoor(get_turf(src))
+				qdel(src)
+				return
+		if(CONSTRUCTION_WIRES_EXPOSED)
+			if(istype(C, /obj/item/weapon/wirecutters))
+				playsound(get_turf(src), 'sound/items/Wirecutter.ogg', 50, 1)
+				user.visible_message("<span class='notice'>[user] starts cutting the wires from [src]...</span>", \
+									 "<span class='notice'>You begin removing [src]'s wires...</span>")
+				if(!do_after(user, 60)) return

```

Comment:
Target argument is needed. Plus a constructionStep check to prevent duplicating cable coil.